product_state/views/product_views.xml

```diff
@@ -1,16 +1,122 @@
+                    </group>
+                </sheet>
+            </form>
+        </field>
+    </record>
+
+    <record id="view_product_state_tree" model="ir.ui.view">
+        <field name="name">product.state.tree</field>
+        <field name="model">product.state</field>
+        <field name="arch" type="xml">
+            <tree string="product.state">
+                <field name="name"/>
+                <field name="description"/>
+            </tree>
+        </field>
+    </record>
+
+    <record id="view_product_state_kanban" model="ir.ui.view">
+        <field name="name">product.state.kanban</field>
+        <field name="model">product.state</field>
+        <field name="arch" type="xml">
+            <kanban>
+                <field name="id"/>
+                <field name="products_count"/>
+                <field name="description"/>
+                <templates>
+                    <t t-name="kanban-box">
+                        <div class="oe_kanban_global_click">
+                            <div class="oe_kanban_details">
+                                <h4>
+                                    <field name="name"/>
+                                </h4>
+                                <div>
+                                    <a name="%(product_state.action_open_state_products)d" type="action">
+                                        <t t-esc="record.products_count.value"/> Products
+                                    </a>
+                                </div>
+                            </div>
+                        </div>
+                    </t>
+                </templates>
+            </kanban>
+        </field>
+    </record>
+
+    <record id="view_product_template_search_state" model="ir.ui.view">
+        <field name="name">product.template.search.state</field>
+        <field name="model">product.template</field>
+        <field name="inherit_id" ref="product.product_template_search_view"/>
+        <field name="arch" type="xml">
+            <field name="name" position="after">
+                <field name="product_state_id"/>
+                <filter string="State" name="groupby_state" domain="[]"
+                        context="{'group_by' : 'product_state_id'}"/>
+                <separator/>
+            </field>
+        </field>
+    </record>
 
     <record id="product_template_form_view" model="ir.ui.view">
         <field name="name">product.template.common.form</field>
         <field name="model">product.template</field>
         <field name="inherit_id" ref="product.product_template_form_view"/>
         <field name="arch" type="xml">
-            <sheet position="before">
-                <header>
-                    <field name="state" widget="statusbar" clickable="True"/>
-                </header>
-            </sheet>
+            <header position="inside">
+                <field name="product_state_id" widget="statusbar" clickable="True"/>
```

Comment:
I see on runbot that for new records does not work default, so you need to add filed state on form
<field name="state" invisible="1"/>

Reply:
done